File: CrmApp.Client/src/models/DTO/contractors/ContractorListItemDTO.ts
import { AdditionalFieldValueDTO } from "../AdditionalFieldValueDTO";

export class ContractorListItemDTO {
    id: number = 0;
    name: string = '';
    code: string = '';
    displayName: string = '';
    nip?: string = '';   
    isXopero?: boolean = false;
    activeEngagementTypes: number[] = []
    hoursLimit?: number;
    hoursUsed?: number
    hoursRemaining?: number;
    additionalFieldValues: AdditionalFieldValueDTO[] = [];
}

File: CrmApp.Domain/DTO/ContractorsDTO/ContractorListItemDTO.cs
using CrmApp.Domain.DTO.AdditionalFields;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.DTO.ContractorsDTO
{
    public class ContractorListItemDTO
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public string DisplayName { get; set; } = string.Empty;
        public string? Nip { get; set; }
        public bool? IsXopero { get; set; }

        public List<int> ActiveEngagementTypes { get; set; } = [];

        public decimal? HoursLimit { get; set; }
        public decimal? HoursUsed { get; set; }
        public decimal? HoursRemaining { get; set; }

        public List<AdditionalFieldValueDTO> AdditionalFieldValues { get; set; } = [];
    }
}
